share/eds/retriever.go

```diff
@@ -104,23 +104,18 @@ func (r *Retriever) Retrieve(ctx context.Context, dah *da.DataAvailabilityHeader
 // quadrant request retries. Also, provides an API
 // to reconstruct the block once enough shares are fetched.
 type retrievalSession struct {
+	dah   *da.DataAvailabilityHeader
 	bget  blockservice.BlockGetter
 	adder *ipld.NmtNodeAdder
 
-	treeFn rsmt2d.TreeConstructorFn
-	codec  rsmt2d.Codec
-
-	dah            *da.DataAvailabilityHeader
-	squareImported *rsmt2d.ExtendedDataSquare
-
-	quadrants   []*quadrant
-	sharesLks   []sync.Mutex
-	sharesCount uint32
-
-	squareLk  sync.RWMutex
-	square    [][]byte
-	squareSig chan struct{}
-	squareDn  chan struct{}
+	// TODO(@Wondertan): Extract into a separate data structure
```

Comment:
[non-blocking] why? Also we should probably refrain from having TODOs like this unless there is also a relevant issue open (but i am also guilty of this)

Reply:
It's about [this](https://github.com/celestiaorg/rsmt2d/issues/135) issue. The original idea was to just extract a thread-safe EDS within the node repo, but rsmt2d authors actually want to have it there.

Reply:
I can link it there btw